src/credential.rs

```diff
@@ -101,13 +228,20 @@ pub struct CredentialData {
     pub cred_protect: Option<CredentialProtectionPolicy>,
     // TODO: add `sig_counter: Option<CounterId>`,
     // and grant RKs a per-credential sig-counter.
+
+    // In older app versions, we serialized the full credential to determine the credential ID.  In
+    // newer app versions, we strip unnecessary fields to generate a shorter credential ID.  To
+    // make sure that the credential ID does not change for an existing credential, this field is
+    // used as a marker for new credentials.
+    #[serde(default)]
+    use_short_id: bool,
```

Comment:
Should not be serialized if `false`.

Reply:
`skip_serializing_if` cannot be used here due to https://github.com/ycrypto/serde-indexed/issues/2.  Wrapping the field in an `Option` works.